components/FilterPlants/FilterPlants.js

```diff
@@ -24,12 +52,51 @@ export default function FilterPlants({
         {lightOptions.map((option) => (
           <div key={option.id}>
             <input
-              type="radio"
+              type="checkbox"
               id={option.id}
               name="lightNeed"
               value={option.value}
-              checked={selectedFilter === option.value}
-              onChange={(event) => onFilterValue(event.target.value)}
+              checked={selectedFilter.lightNeed === option.value}
+              onChange={handleCheckbox}
+            />
+            <label htmlFor={option.id}>{option.label}</label>
+          </div>
+        ))}
+      </section>
+
+      <label>
+        <h3>Water Needs:</h3>
+      </label>
+      <section>
+        {waterOptions.map((option) => (
+          <div key={option.id}>
+            <input
+              type="checkbox"
+              id={option.id}
+              name="waterNeed"
+              value={option.value}
+              checked={selectedFilter.waterNeed === option.value}
+              onChange={handleCheckbox}
+            />
+            <label htmlFor={option.id}>{option.label}</label>
+          </div>
+        ))}
+      </section>
```

Comment:
The `lightOptions`, `waterOptions`, and` fertiliserOptions` sections repeat the same logic for rendering checkboxes.

➡️ Consider to extracting the repeated rendering logic into a reusable component.

```js
function FilterSection({ title, options, name, selectedValue, onChange }) {
  return (
    <section>
      <h3>{title}</h3>
      {options.map((option) => (
        <div key={option.id}>
          <input
            type="checkbox"
            id={option.id}
            name={name}
            value={option.value}
            checked={selectedValue === option.value}
            onChange={onChange}
          />
          <label htmlFor={option.id}>{option.label}</label>
        </div>
      ))}
    </section>
  );
}


// Usage Example
<FilterSection
  title="Light Needs"
  options={lightOptions}
  name="lightNeed"
  selectedValue={selectedFilter.lightNeed}
  onChange={handleCheckbox}
/>


```